pkg/drain/drainer_test.go

```diff
@@ -0,0 +1,255 @@
+		It("should failed if the platform is not able to drain", func() {
+			n, _ := createNode("node0")
+			platformHelper.EXPECT().OpenshiftBeforeDrainNode(ctx, n).Return(false, fmt.Errorf("failed"))
+
+			completed, err := drn.DrainNode(ctx, n, false, false)
+			Expect(err).To(HaveOccurred())
+			Expect(completed).To(BeFalse())
+		})
+
+		It("should return not completed base on OpenshiftBeforeDrainNode call", func() {
+			n, _ := createNode("node0")
+			platformHelper.EXPECT().OpenshiftBeforeDrainNode(ctx, n).Return(false, nil)
+
+			completed, err := drn.DrainNode(ctx, n, false, false)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(completed).To(BeFalse())
+		})
+
+		It("should return error if not able to cordon the node", func() {
+			n, _ := createNode("node0")
+			nCopy := n.DeepCopy()
+			nCopy.Name = "node1"
+			originalDrainTimeOut := drain.DrainTimeOut
+			drain.DrainTimeOut = 3 * time.Second
+			defer func() {
+				drain.DrainTimeOut = originalDrainTimeOut
+			}()
+
+			platformHelper.EXPECT().OpenshiftBeforeDrainNode(ctx, nCopy).Return(true, nil)
+
+			_, err := drn.DrainNode(ctx, nCopy, false, false)
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("should return error if not able to drain the node", func() {
+			n, _ := createNode("node0")
+			platformHelper.EXPECT().OpenshiftBeforeDrainNode(ctx, n).Return(true, nil)
+			createPodWithFinalizerOnNode(ctx, "test-node-0", "node0")
+			originalDrainTimeOut := drain.DrainTimeOut
+			drain.DrainTimeOut = 3 * time.Second
+			defer func() {
+				drain.DrainTimeOut = originalDrainTimeOut
+			}()
+
+			_, err := drn.DrainNode(ctx, n, true, false)
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("should remove pods with sriov devices only if a full drain is not requested", func() {
+			n, _ := createNode("node1")
+			resources := map[corev1.ResourceName]resource.Quantity{corev1.ResourceName(fmt.Sprintf("%s/test", vars.ResourcePrefix)): resource.MustParse("1")}
+			n.Status.Allocatable = resources
+			n.Status.Capacity = resources
+			err := k8sClient.Status().Update(ctx, n)
+			Expect(err).ToNot(HaveOccurred())
+			platformHelper.EXPECT().OpenshiftBeforeDrainNode(ctx, n).Return(true, nil)
+			createPodOnNode(ctx, "test-node-1", "node1")
+			createPodWithSriovDeviceOnNode(ctx, "test1-node-1", "node1")
+
+			go func() {
+				Eventually(func(g Gomega) {
+					podObj := &corev1.Pod{}
+					err = k8sClient.Get(ctx, client.ObjectKey{Name: "test1-node-1", Namespace: testNamespace}, podObj)
+					g.Expect(err).ToNot(HaveOccurred())
+					g.Expect(podObj.DeletionTimestamp).ToNot(BeNil())
+					err = k8sClient.Delete(ctx, podObj, &client.DeleteOptions{GracePeriodSeconds: pointer.Int64(0)})
+					g.Expect(err).ToNot(HaveOccurred())
+				}, 2*time.Minute, time.Second).Should(Succeed())
+			}()
+
+			_, err = drn.DrainNode(ctx, n, false, false)
+			Expect(err).ToNot(HaveOccurred())
+			pod := &corev1.Pod{}
+			err = k8sClient.Get(ctx, client.ObjectKey{Name: "test-node-1", Namespace: testNamespace}, pod)
```

Comment:
`test-node-1` and `test1-node-1` are pretty similar names. Can you switch to something like:

`regularPod`/`sriovPod1` o even a simple `podBlue`/`podGreen`

?

Reply:
sure done